lib/providers/token-price/index.ts

```diff
@@ -0,0 +1,82 @@
+import { BigNumber, ethers } from "ethers";
+import { default as Logger } from 'bunyan';
+import { Token } from "@uniswap/sdk-core";
+import { USDC_ON } from "@uniswap/smart-order-router";
+import axios from "axios";
+
+export type BucketRange = {
+    lower: BigNumber;
+    upper: BigNumber;
+}
+
+export type TokenAmountsBucket = {
+    bucketRange: BucketRange 
+    lower: BigNumber;
+    upper: BigNumber;
```

Comment:
maybe call these `lowerInUSDC`, `upperInUSDC` for better clarity

Reply:
not doing the provider anymore